test/pipelinerun_test.go

```diff
@@ -175,15 +176,23 @@ func TestPipelineRun(t *testing.T) {
 			if len(events) != td.expectedNumberOfEvents {
 				t.Fatalf("Expected %d number of successful events from pipelinerun and taskrun but got %d; list of receieved events : %#v", td.expectedNumberOfEvents, len(events), events)
 			}
-			// Check to make sure the PipelineRun's artifact storage PVC has been "deleted" at the end of the run.
-			pvc, err := c.KubeClient.Kube.CoreV1().PersistentVolumeClaims(namespace).Get(artifacts.GetPVCName(pipelineRun), metav1.GetOptions{})
-			if err != nil {
+
+			// Wait for up to 10 minutes and restart every 30seconds to check if
+			// the PersistentVolumeClaims has the DeletionTimestamp
+			if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+				// Check to make sure the PipelineRun's artifact storage PVC has been "deleted" at the end of the run.
+				pvc, errWait := c.KubeClient.Kube.CoreV1().PersistentVolumeClaims(namespace).Get(artifacts.GetPVCName(pipelineRun), metav1.GetOptions{})
+				if errWait != nil && !errors.IsNotFound(errWait) {
+					return true, fmt.Errorf("Error looking up PVC %s for PipelineRun %s: %s", artifacts.GetPVCName(pipelineRun), prName, errWait)
+				}
+				return pvc.DeletionTimestamp == nil, errWait
+			}); err != nil {
 				if !errors.IsNotFound(err) {
```

Comment:
:thinking: shouldn't that check be in the `PollImmediate` ? (and returning `true` + `nil` as error) so that we don't have to do the check twice ?